### ISSUE TITLE
Why does the tabpane get new tabs in this example?

### ISSUE THREAD

**Issue by ysm-coder** [edvin/tornadofx#1148]
Hello, I maged it to reproduce a problem I have in a simplefied version. Can you pls explain me why the labels get added as tabs to the tabpane and how I can avoid this? (just run the program and click on the button If you dont get what I mean):

```
import javafx.scene.layout.VBox
import tornadofx.*

class ExampleView : View("My View") {
    lateinit var theVBox :VBox

    override val root = tabpane {
        tab("the only tab I want") {
            borderpane {
                center(){
                    button("add label") {
                        action{
                            x()
                        }
                    }
                }
                bottom(){
                    theVBox = vbox {  }
                }
            }
        }
    }

    private fun x(){
        theVBox.add(
            label("test")
        )
    }
}
```

Putting the tabpane into a vbox seems to be a workaround for this but it gives me other Problems.

**Comment by edvin** [comment]
You must remember/understand that builders operate _on_ the object you call them on. Calling `label()` is not just creating a label, it is also automatically binding it to the parent list on the object you called it on. There is also a special case for the View itself, so you can call builders on the view to have them added as the root element. That's what's happening in your `x()` function. When you write `theVBox.add(label("test"))`, you are trying to add the label both to the vbox and to the view root. Just calling `theVBox.label("test")` should solve your problem.

**Comment by ysm-coder** [comment]
> You must remember/understand that builders operate on the object you call them on.

Yes, I was searching my bug somewhere else but after I assumed this, I could reproduce it and create this example here. But I didnt know how to not do it.


> Just calling `theVBox.label("test")`

So in contrast to `.add()` `.label()` is not a builder? 


> should solve your problem

Yes it did, thank you for your help and for tornadofx itself.

**Comment by edvin** [comment]
It's the other way around: `label()` is a builder, `add()` is not. `add()` simply adds whatever you pass in to the childer list of the element you called `add` on. With builders, you don't need to use `add`, as it's done automatically for you.

The `alert.dialogPane.content` code you refer to looks very strange. You shouldn't be doing that. Why would you assign the content of a dialogPane to an hbox you build and assign to another view? This makes no sense to me. Perhaps if you try to explain what you're trying to accomplish, I can provide you with a code sample to do what you need?

**Comment by ysm-coder** [comment]
I do not try to assign it to another view, actually I have just one view (MainView). I moved some functions from my MainView class into another new Kotlin file to keep Mainview more clean. So from MainView I had called that function in the other File to show the allert, but to use `.hbox{}` in that function I needed an Object to call it on (Im just interested in the HBox it returns as content to my dialogPane which haves the content I specify inside the `{}` ). First I gave the MainView as an parameter to that function and called `.hbox{}` on it, this worked as long as my root was a vbox, but with tabpane as the root it added the HBox as a tab to my root (you already explained why this happens). I also had tried `alert.dialogPane.content.hbox{}` but this gave me an exception If Im remembering correctly. Just to make it clear, despite beeing a workaround, `alert.dialogPane.content = Region().hbox` works at the moment for me, but Im creatng a `Region()` just to be able to create a `HBox()` with `.hbox{}`
